content/blog/2024-05-22-rustl-2024-unconference.md

```diff
@@ -0,0 +1,196 @@
+
+The requests were well-documented, with motivating examples in existing Dioxus code that would clearly be improved by the features.
+Most of them felt both self-contained and like they would tremendously help beginners working on their first Rust project.
+
+Overall people were very enthusiastic about them, and the lang team members in the room seemed pretty optimistic that those features could be added.
+
+My favorite proposals were:
+
+- `Capture` trait for auto-cloning reference-counted types in closures.
+- Automatic partial borrows.
+- Optional struct args.
+
+Jon also wanted support for a remote cache of pre-built crates, so that users wouldn't suffer from having to rebuild dozens of crates every time you download your crate's dependencies.
+The consensus from Rust maintainers was that such a cache would be essentially impossible with Rust's current architecture.
+
+Niko Matsakis recently released [The borrow checker within](https://smallcultfollowing.com/babysteps/blog/2024/06/02/the-borrow-checker-within/), which feels like a follow-up to some of these discussions.
+
+
+## Collaboration in the Rust GUI ecosystem
+
+Over the last two years I've joked a few times that everybody in the Rust GUI ecosystem wants the ecosystem to standardize on common crates, as long as it's *their* crates.
+
+After RustNL, I'm pleased to report this joke isn't accurate anymore.
+
+A vibe that permeated all discussions was people wanting work to be pooled.
+It felt like people didn't strongly care whose crate we adopted, as long as we all agreed the crate made sense:
+we're slowly crawling out of the xkcd 927 dilemma.
+
+The main ecosystem crates that people are enthusiastic about sharing are:
+
+- [winit](https://github.com/rust-windowing/winit) for creating windows.
+- [AccessKit](https://github.com/AccessKit/accesskit/) for plugging into accessibility APIs.
+- [Parley](https://github.com/linebender/parley) for rendering text and as a basis for a text-editing widget.
+
+Jon Kelley also said he would be interested in using Masonry as the native backend for Dioxus, though we haven't hashed out details yet.
+
+### Corporate funding
+
+For open-source projects, "Who funds this" is a difficult question: in any healthy project, there is a large scale of contributions, with individual non-corporate contributors at one end, and companies paying cash to the project's treasury at the other.
+In-between are self-employed people like me contracted to work on an open-source project, and corporate employees who contribute to the project as part of their 9-to-5 job.
+
+Some monetary contributions can also come from individual non-corporate donors: Servo has about fifty of them, for instance.
+
+With that in mind, some notable sponsors for projects represented at the Unconference were:
+
+- **Google Fonts:** Linebender projects.
+- **Futurewei:** Dioxus, Makepad, and Servo.
```

Comment:
Kindly add Robius here, which is the primary funding objective from Futurewei.

Note that Futurewei also funds WyeWorks' work on Makepad; not sure if that's relevant.